nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
I agree that it's still possible to get multiple instances in the example you mention, but in this case I think it's acceptable, since it would be a very rare and strange thing for the user to both add `new RequiredFieldValidator()` and also use `LabeledFieldController#setRequired`. Typically, the user would only use one approach or the other. In the unusual case that the user does do it, the only side-effect is that they would see duplicate error messages, which isn't a huge problem as it would be detected easily at runtime.